sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -0,0 +1,118 @@
+
+    :param source: Required. Identifies the context in which an event happened. The combination of id and source must
+     be unique for each distinct event. If publishing to a domain topic, source must be the domain name.
+    :type source: str
+    :param type: Required. Type of event related to the originating occurrence.
+    :type type: str
+    :keyword data: Optional. Event data specific to the event type. If data is of bytes type, it will be sent
+     as data_base64 in the outgoing request.
+    :type data: object
+    :keyword time: Optional. The time (in UTC) the event was generated, in RFC3339 format.
+    :type time: ~datetime.datetime
+    :keyword dataschema: Optional. Identifies the schema that data adheres to.
+    :type dataschema: str
+    :keyword datacontenttype: Optional. Content type of data value.
+    :type datacontenttype: str
+    :keyword subject: Optional. This describes the subject of the event in the context of the event producer
+     (identified by source).
+    :type subject: str
+    :keyword specversion: Optional. The version of the CloudEvent spec. Defaults to "1.0"
+    :type specversion: str
+    :keyword id: Optional. An identifier for the event. The combination of id and source must be
+     unique for each distinct event. If not provided, a random UUID will be generated and used.
+    :type id: Optional[str]
+    :ivar source: Identifies the context in which an event happened. The combination of id and source must
+     be unique for each distinct event. If publishing to a domain topic, source must be the domain name.
+    :vartype source: str
+    :ivar data: Event data specific to the event type.
+    :vartype data: object
+    :ivar type: Type of event related to the originating occurrence.
+    :vartype type: str
+    :ivar time: The time (in UTC) the event was generated, in RFC3339 format.
+    :vartype time: ~datetime.datetime
+    :ivar dataschema: Identifies the schema that data adheres to.
+    :vartype dataschema: str
+    :ivar datacontenttype: Content type of data value.
+    :vartype datacontenttype: str
+    :ivar subject: This describes the subject of the event in the context of the event producer
+     (identified by source).
+    :vartype subject: str
+    :ivar specversion: Optional. The version of the CloudEvent spec. Defaults to "1.0"
+    :vartype specversion: str
+    :ivar id: An identifier for the event. The combination of id and source must be
+     unique for each distinct event. If not provided, a random UUID will be generated and used.
+    :vartype id: Optional[str]
+    """
+    def __init__(self, source, type, **kwargs): # pylint: disable=redefined-builtin
+        # type: (str, str, Any) -> None
+        self.source = source
+        self.type = type
+        self.specversion = kwargs.pop("specversion", "1.0")
+        self.id = kwargs.pop("id", str(uuid.uuid4()))
+        self.time = kwargs.pop("time", dt.datetime.utcnow().isoformat())
```

Comment:
https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow
Warning Because naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. As such, the recommended way to create an object representing the current time in UTC is by calling datetime.now(timezone.utc).

Problem is, Python 2 doesn't have timezone.utc. @johanste usually I tell people to use msrest.serialization.utc since they use msrest as a dependency anyway, but here it won't work. Though?

Reply:
updated